### PR TITLE
fix(Header): Improve styling of header buttons

### LINKED ISSUE #376: Active Header action has no underline
#### Describe the bug (actual behavior)
Both HeaderButton & IconButton components have no underline when `isActive={true}` is set. 

#### Expected behavior
Active item should be underlined same as it would be underlined using HeaderButton in breadcrumbs.

#### Reproduction
Link to a minimal repro: https://codesandbox.io/s/itwinui-react-active-header-element-highlight-issue-1024k

##### Steps to reproduce
1. Use HeaderButton/IconButton as Header actions.
1. Set any of them as active.

#### Additional information
Used this as a temp workaround:
```
.iui-button.iui-active,
.iui-header-button.iui-active {
  box-shadow: 0 2px 0 0 var(--iui-color-foreground-primary);
}
```

### PR DESCRIPTION
**Pull request by mayank99**
1. Moved `iui-header-buttons` mixin out from `iui-left`. This allows header buttons to be used in `iui-right` with the active stripe (fixes #376)
2. Added ability to render header-logo as `button` or `a` element.
3. Added explicit `content-box` sizing to `iui-left` so that it doesn't get reset by user's stylesheet.
4. Fixed random bugs in split header button re: responsive design and disabled icon fill

Added scenarios and images for visually testing responsive design.

### CODE REVIEW

File: src/header/header.scss [hunk 1]
```diff
@@ -13,8 +13,7 @@
   user-select: none;
```
Comment (bentleyvk) [inline src/header/header.scss:13]: Could you undo formatting because it is hard to review?

Comment (mayank99) [reply src/header/header.scss:13]: That's why I created #372

Comment (mayank99) [reply src/header/header.scss]: Done. Also I suggest hiding whitespace changes in git diff

File: src/header/header.scss [hunk 2]
```diff
@@ -161,45 +162,62 @@
     @include iui-header-button-icon;
   }
 
-  .iui-header-button {
+  .iui-header-button.iui-header-button { // boost specificity
```
Comment (bentleyvk) [inline src/header/header.scss:165]: Why this change is needed?

Comment (mayank99) [reply src/header/header.scss:165]: Because I moved the class out of `iui-left`, the specificity was lowered and was breaking styling, so I had to boost it back up

File: src/header/header.scss [hunk 3]
```diff
@@ -119,9 +120,9 @@
 
   @media (max-width: 768px) {
     &,
-    &.iui-slim.iui-slim {
+    &.iui-slim {
       .iui-left {
-        .iui-header-button {
+        .iui-button {
```
Comment (bentleyvk) [inline src/header/header.scss:125]: Why changed to target all buttons?

Comment (mayank99) [reply src/header/header.scss:125]: I noticed it was not working for split header button (which does not use `iui-header-button` class)

Comment (bentleyvk) [reply src/header/header.scss:125]: Maybe add that to test HTML.

Comment (mayank99) [reply src/header/header.scss:125]: huh? it's already there

Comment (bentleyvk) [reply src/header/header.scss:125]: Then why I don't see any difference? 😁

Comment (mayank99) [reply src/header/header.scss:125]: this is inside a media query, it only happens on small screens

File: src/header/header.scss (hunk 2 above)
Comment (FlyersPh9) [reply src/header/header.scss:165]: Will this boost in specificity be needed after #363 passes?

**Comment by FlyersPh9** [comment]
Glad you were able to fix the disabled dropdown arrow bug, I tried fixing that a couple weeks back and was unsuccessful.

**Review by FlyersPh9 — APPROVED** [review]

File: src/header/header.scss (hunk 2 above)
Comment (mayank99) [reply src/header/header.scss:165]: Hopefully not, but we'll see
